core/src/components/button/button.tsx

```diff
@@ -158,9 +164,10 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
 
     } else if (hasShadowDom(this.el)) {
       // this button wants to specifically submit a form
-      // climb up the dom to see if we're in a <form>
+      // if form property is defined, it will be used to submit
+      // else climb up the dom to see if we're in a <form>
       // and if so, then use JS to submit it
-      const form = this.el.closest('form');
+      const form = (this.form) ? this.form : this.el.closest('form');
```

Comment:
Since the `form` prop needs to be an ID, we will need to do additional work here to select the right form. I have not tested this, but I imagine the helper function would look something like this:

```js
/**
  * Given a reference element, find the closest form.
  * If there is no reference element, search the document.
  * If `id` is provided, the selector will only look for forms
  * given that id.
  */
const getForm = (refEl: HTMLElement, id?: string) => {
  const selector = (id) ? `form#{id}` : 'form';
  if (refEl) {
    const form = refEl.closest(selector);
    if (form) return form;
  }

  if (typeof (document as any) === 'undefined') return null;
        
  return document.querySelector(selector);
}

const form = getForm(this.el, this.form);
```

(The `document` check is done to make sure this does not break if this happens to get called when doing server side rendering. I don't think it will, but let's add it just to be safe)

Reply:
Thanks for the helper function. It did the work